homeassistant/components/leviosa_shades/cover.py

```diff
@@ -0,0 +1,136 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import DeviceInfo
+
+from .const import (
+    BLIND_GROUPS,
+    DOMAIN,
+    MANUFACTURER,
+    MODEL,
+    SERVICE_NEXT_DOWN_POS,
+    SERVICE_NEXT_UP_POS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Estimated time it takes to complete a transition
+# from one state to another
+TRANSITION_COMPLETE_DURATION = 30
+PARALLEL_UPDATES = 1
+
+COVER_NEXT_POS_SCHEMA = vol.Schema({vol.Optional(ATTR_ENTITY_ID): cv.entity_ids})
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Leviosa shade groups."""
+    _LOGGER.debug(
+        "Setting up %s[%s]: %s",
+        entry.domain,
+        entry.title,
+        entry.entry_id,
+    )
+    hub_name = entry.title
+    hub_mac = entry.data["device_mac"]
+    hub_ip = entry.data["host"]
+    blind_groups = entry.data[BLIND_GROUPS]
+    _LOGGER.debug("Groups to create: %s", blind_groups)
+    hub = tZoneHub(
+        hub_ip=hub_ip, hub_name=hub_name, websession=async_get_clientsession(hass)
+    )
+    await hub.getHubInfo()  # Check all is good
+    _LOGGER.debug("Hub object created, FW: %s", hub.fwVer)
+    entities = []
+    for blind_group in blind_groups:
+        _LOGGER.debug("Adding blind_group: %s", blind_group)
+        new_group_obj = hub.AddGroup(blind_group)
+        entities.append(
+            LeviosaBlindGroup(
+                hass, hub_mac + "-" + str(new_group_obj.number), new_group_obj
+            )
+        )
+    async_add_entities(entities)
+
+    _LOGGER.debug("Setting up Leviosa shade group services")
+    platform = entity_platform.current_platform.get()
+    platform.async_register_entity_service(
+        SERVICE_NEXT_DOWN_POS,
+        COVER_NEXT_POS_SCHEMA,
+        "next_down_position",
+    )
+    platform.async_register_entity_service(
+        SERVICE_NEXT_UP_POS,
+        COVER_NEXT_POS_SCHEMA,
+        "next_up_position",
+    )
+
+
+class LeviosaBlindGroup(cover.CoverEntity):
+    """Represents a Leviosa shade group entity."""
+
+    def __init__(
+        self, hass: HomeAssistant, blind_group_id, blind_group_obj: tShadeGroup
+    ) -> None:
+        """Initialize the shade group."""
+        self._blind_group_id = blind_group_id
+        self._blind_group_obj = blind_group_obj
+        self._hass = hass
+
+        self._attr_name = self._blind_group_obj.name
+        self._attr_unique_id = self._blind_group_id
+        self._attr_device_class = cover.CoverDeviceClass.SHADE
+        self._attr_supported_features = (
+            cover.CoverEntityFeature.OPEN
+            | cover.CoverEntityFeature.CLOSE
+            | cover.CoverEntityFeature.STOP
+        )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._blind_group_obj.Hub.hub_ip)},
```

Comment:
My understanding is that this is only a LAN device and does not have a cloud service, so all communication depends on the correct IP. If the IP changes, it can only be re-binded. Unfortunately, leviosa does not provide a cloud service.
So we can only continue to use IP here.